src/app/child-dev-project/children/child-photo-service/datatype-photo.ts

```diff
@@ -20,31 +20,42 @@ import { ChildPhotoService } from "./child-photo.service";
 import { EntitySchemaField } from "../../../core/entity/schema/entity-schema-field";
 import { EntitySchemaService } from "../../../core/entity/schema/entity-schema.service";
 import { Entity } from "../../../core/entity/entity";
+import { Photo } from "./photo";
+import { SafeUrl } from "@angular/platform-browser";
+import { BehaviorSubject } from "rxjs";
 
 /**
  * Dynamically load the child's photo through the ChildPhotoService during Entity loading process.
  */
-export class LoadChildPhotoEntitySchemaDatatype
-  implements EntitySchemaDatatype {
-  public readonly name = "load-child-photo";
+export class PhotoDatatype implements EntitySchemaDatatype {
+  public readonly name = "photo";
 
-  constructor(private childPhotoService: ChildPhotoService) {}
-
-  public transformToDatabaseFormat(value) {
-    return undefined;
+  public transformToDatabaseFormat(value: Photo, schema: EntitySchemaField) {
+    if (value.path === schema.defaultValue) {
```

Comment:
we could delete the old `photoFile` property here during save and thereby gradually migrate documents, whenever a user updates them

Reply:
Deleting is not necessary: The old `photoFile` attribute does not have a schema anymore and is therefore not saved to the database.